File: common/csharp_common/json_manipulate.cs
// --------------------------------------------------------------------
/*
	json_manipulate.cs

				Feb/27/2012


*/
// --------------------------------------------------------------------
using	System;
using	System.IO;
using	System.Collections.Generic;

using	Newtonsoft.Json;

// --------------------------------------------------------------------
public class	json_manipulate
{
// --------------------------------------------------------------------
public static Dictionary <string,Object>
	json_to_dict_proc (String str_json)
{
	Dictionary <string,Object> dict_aa
			= new Dictionary <string,Object> ();

	Dictionary <string,Object> dict_bb
		= JsonConvert.DeserializeObject
		<Dictionary <string,Object>> (str_json);

	foreach (KeyValuePair<string, Object> kv in dict_bb)
		{
		if ((kv.Key != "_id") && (kv.Key != "_rev"))
			{
			Dictionary <string,string> unit_aa
				= JsonConvert.DeserializeObject
				<Dictionary <string,string>> (kv.Value.ToString ());

			dict_aa = text_manipulate.dict_append_proc
				(dict_aa,kv.Key,unit_aa["name"],
					int.Parse (unit_aa["population"]),
					unit_aa["date_mod"]);
			}
		else
			{
			dict_aa[kv.Key] = kv.Value;
			}
		}


	return	dict_aa;
}

// --------------------------------------------------------------------
}

// --------------------------------------------------------------------

File: common/fsharp_common/json_manipulate.fs
// --------------------------------------------------------------------
//
//	json_manipulate.fs
//
//				Mar/09/2018
//
//
//
// --------------------------------------------------------------------
namespace json_manipulate
open System
open System.IO
open System.Text
open System.Collections.Generic

open Newtonsoft.Json
// --------------------------------------------------------------------
module m01 =
  let json_to_dict_proc (str_json:string) =
//    let dict_aa = new Dictionary<string,Dictionary<string,string>>()
    let dict_aa = JsonConvert.DeserializeObject<Dictionary<string,Dictionary<string,string>>> (str_json)
    dict_aa
// --------------------------------------------------------------------
  let file_write_proc (file_name:string,str_out:string) =
    0
// --------------------------------------------------------------------
